### ISSUE TITLE
Setting a Gf from a Gf with a different beta should be protected, and not overwrite the mesh

### ISSUE THREAD

**Issue by the-hampel** [TRIQS/triqs#772]
### Description

The beta of a mesh of a Gf will be overwritten if set from a Gf at another temperature without warnings or errors. I think this should be protected to prevent errors by checking that the beta of the mesh matches, otherwise the high-frequency behavior of the Gf is very unreasonable.

### Steps to Reproduce

**Expected behavior:** 
```
mesh_b40 = MeshImFreq(beta=40, S="Fermion", n_max=1025)
Giw_b40 = Gf(mesh= mesh_b40, target_shape=[])

mesh_b20 = MeshImFreq(beta=20, S="Fermion", n_max=1025)
Giw_b20 = Gf(mesh= mesh_b20, target_shape=[])

Giw_b20 << Giw_b40
```
shoud give something like:
```
AssertionError: Meshes are incompatible beta=20 vs beta=40
```

**Actual behavior:** 
```
mesh_b40 = MeshImFreq(beta=40, S="Fermion", n_max=1025)
Giw_b40 = Gf(mesh= mesh_b40, target_shape=[])

mesh_b20 = MeshImFreq(beta=20, S="Fermion", n_max=1025)
Giw_b20 = Gf(mesh= mesh_b20, target_shape=[])

Giw_b20 << Giw_b40

print Giw_b20.mesh
```
gives 
```
Matsubara Freq Mesh of size 2050, Domain: Matsubara domain with beta = 40, statistic = Fermion, positive_only : 0
```
### Versions
triqs `2.2.x , unstable`

**Comment by aichhorn** [comment]
There could be another meaning to Giw_b20 << Giw_b40. I am not sure if this is intended, but this could also imply an interpolation/extrapolation of the Greens function to a different beta. This would be useful in quite some occasions, but I don't know if this is the actual behavior.
The high frequency expansion should not depend on the mesh, since it is for given data an analytic function of iw, not matter at which iw it is evaluated. If the data is not extrapolated but copied, the tail is wrong of course.

**Comment by the-hampel** [comment]
I can confirm, that in the current implementation the data is (as far as I can see it) copied, and the tails are wrong. I tried the above operation with a Gf where the first moment of the tail is 1 and afterwards when going from b30 to b20 the moment was 0.66. That was actually how I noticed it, because I accidentally loaded a Sigma from a different beta into my calculation and the solver had huge troubles, but did not give any warning. 

I agree that interpolating from one beta to another would be nice to have!
